app/components/nav-link-to.js

```diff
@@ -5,16 +5,22 @@ export default LinkComponent.extend({
 
   tagName: 'li',
 
-  hrefForA: computed('models', 'qualifiedRouteName', function computeLinkToComponentHref() {
-    let route = this.route;
-    let models = this.models;
+  hrefForA: computed(
+    '_currentRouterState',
+    '_route',
+    '_models',
+    '_query',
+    'tagName',
+    'loading',
+    'loadingHref',
+    function computeLinkToComponentHref() {
+      if (this.loading) {
+        return this.loadingHref;
+      }
 
-    if (this.loading) {
-      return this.loadingHref;
-    }
+      const { _route: route, _models: models, _query: query, _routing: routing } = this;
 
-    let routing = this._routing;
-    let queryParams = this.get('queryParams.values');
-    return routing.generateURL(route, models, queryParams);
-  })
+      return routing.generateURL(route, models, query);
```

Comment:
Do we still need to employ private API for this? If we do, we should probably rethink the whole component.

Reply:
This could be a temporary fix until migrating to `ember-bootstrap`. I expect that ember-bootstrap manage active state properly, so we don't have to deal with this issue anymore.